### PR TITLE
Prepare tutorial about binding Service Instances to lambda

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Prepare tutorial about binding Service Instances to lambda

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/7048

### CODE REVIEW

File: docs/serverless-v2/08-03-bind-service-instance.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    metadata:
+      name: $NAME
+      namespace: $NAMESPACE
+    spec:
+      clusterServiceClassExternalName: redis
+      clusterServicePlanExternalName: micro
+      parameters:
+        imagePullPolicy: Always
+    EOF    
+    ```
```
Comment (michal-hudy) [inline docs/serverless-v2/08-03-bind-service-instance.md]: Is the Redis available by default in Kyma? Maybe we should say that you need a ServiceInstance first and if not point to the proper tutorial?

Comment (magicmatatjahu) [reply docs/serverless-v2/08-03-bind-service-instance.md]: By default not. Also we haven't any tutorial for provisioning sample serviceClass in our documentation, so here we need to point to external docs, but we have a tutorial to bind in cluster Redis ServiceClass by Addon, but doc for that is outdated :( So my proposition is write as you wrote, user must firstly have Redis ServiceClass and for that we can add small yaml for provisioning Addon with Redis.

Comment (kazydek) [reply docs/serverless-v2/08-03-bind-service-instance.md]: As discussed with @magicmatatjahu , there will be an additional step on adding yaml for provisioning an Addon with Redis.

File: docs/serverless-v2/08-03-bind-service-instance.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+1. Export these variables:
+
+    ```bash
+    export NAME={LAMBDA_NAME}
+    export NAMESPACE=serverless
+    ```
+
+    > **NOTE:** Lambda takes the name from the Function CR name. The ServiceInstance, ServiceBinding, and ServiceBindingUsage CRs can have different names, but for the purpose of this tutorial, all related resources share a common name defined under the **NAME** variable.
+
```
Comment (kazydek) [inline docs/serverless-v2/08-03-bind-service-instance.md]:
```
2. Provision an Addon CR with the Redis service:

cat <<EOF | kubectl apply -f  -
apiVersion: addons.kyma-project.io/v1alpha1
kind: AddonsConfiguration
metadata:
  name: $NAME
  namespace: $NAMESPACE
spec:
  reprocessRequest: 0
  repositories:
    - url: https://github.com/kyma-project/addons/releases/download/0.11.0/index-testing.yaml
EOF


3. Check if the Addon CR was created successfully. The CR phase should state `Phase: Ready`:

kubectl get addons $NAME -n $NAMESPACE -o=jsonpath="{.status.phase}"

```

File: docs/serverless-v2/08-03-bind-service-instance.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+2. Create a ServiceInstance CR. You will provision and use [Redis](https://redis.io/) with its `micro` plan:
```
Comment (kazydek) [inline docs/serverless-v2/08-03-bind-service-instance.md]:
```suggestion
4. Create a ServiceInstance CR. You will use the provisioned [Redis](https://redis.io/) service with its `micro` plan:
```

Comment (kazydek) [reply docs/serverless-v2/08-03-bind-service-instance.md]: Pls update the numbers below as well.

File: docs/serverless-v2/08-03-bind-service-instance.md (hunk 1 above)
Comment (michal-hudy) [reply docs/serverless-v2/08-03-bind-service-instance.md]: Ok :smile:

File: docs/serverless-v2/08-03-bind-service-instance.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+    > **NOTE:** Read [this](/components/service-catalog/#details-provisioning-and-binding) document to learn more about binding in Kyma.
+
+    - The **spec.parameters.envPrefix.name** field is optional. It adds a prefix to all environment variables injected by a given Secret from the Service Binding to the lambda. In our example, **envPrefix** is `REDIS_`, so all environmental variables will follow the `REDIS_{env}` naming pattern.
+
+    > **TIP:** It is considered good practice to use **envPrefix**. In some cases, lambda must use several instances of a given Service Class, so prefixes allow you to distinguish between instances and make sure that one Secret does not overwrite another one.
+
```
Comment (kazydek) [inline docs/serverless-v2/08-03-bind-service-instance.md]: Could you adjust indents?

image

File: docs/serverless-v2/08-03-bind-service-instance.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        name: $NAME
+      parameters:
+        envPrefix:
+          name: "REDIS_"
+    EOF    
+    ```
+
+    - The **spec.serviceBindingRef** and **spec.usedBy** fields are required. **spec.serviceBindingRef** points to the Service Binding you have just created and **spec.usedBy** points to the lambda. More specifically, **spec.usedBy** refers to the name of the related KService CR (`name: $NAME`) and the cluster-specific [UsageKind CR](https://kyma-project.io/docs/components/service-catalog/#custom-resource-usage-kind) (`kind: knative-service`) that defines how Secrets should be injected to your lambda through the Service Binding.
+
+    > **NOTE:** Read [this](/components/service-catalog/#details-provisioning-and-binding) document to learn more about binding in Kyma.
```
Comment (kazydek) [inline docs/serverless-v2/08-03-bind-service-instance.md]: Actually, we already have a similar note and the link at the beginning of the doc, so let's remove this one as it is a duplicate.

File: docs/serverless-v2/08-03-bind-service-instance.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+    - The **spec.parameters.envPrefix.name** field is optional. It adds a prefix to all environment variables injected by a given Secret from the Service Binding to the lambda. In our example, **envPrefix** is `REDIS_`, so all environmental variables will follow the `REDIS_{env}` naming pattern.
+
+    > **TIP:** It is considered good practice to use **envPrefix**. In some cases, lambda must use several instances of a given Service Class, so prefixes allow you to distinguish between instances and make sure that one Secret does not overwrite another one.
```
Comment (kazydek) [inline docs/serverless-v2/08-03-bind-service-instance.md]:
```suggestion
    > **TIP:** It is considered good practice to use **envPrefix**. In some cases, a lambda must use several instances of a given Service Class. Prefixes allow you to distinguish between instances and make sure that one Secret does not overwrite another one.
```

File: docs/serverless-v2/08-03-bind-service-instance.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+    You should get a result similar to the following details:
+
+    ```bash
+    HOST: hb-redis-micro-0e965585-9699-443f-b987-38bc6af0e416-redis.serverless.svc.cluster.local
+    PORT: 6379
+    REDIS_PASSWORD: 1tvDcINZvp
+    ```
+
+    > **NOTE:** If you added the **REDIS_** prefix for environmental variables in step 6, all variables will start with it. For example, the **PORT** variables will take the form of **REDIS_PORT**.
```
Comment (kazydek) [inline docs/serverless-v2/08-03-bind-service-instance.md]:
```suggestion
    > **NOTE:** If you added the **REDIS_** prefix for environmental variables in step 6, all variables will start with it. For example, the **PORT** variable will take the form of **REDIS_PORT**.
```

**Pull request merged**